kore/src/Kore/Step/Simplification/Pattern.hs

```diff
@@ -89,7 +116,36 @@ simplify sideCondition pattern' =
                 & Condition.fromPredicate
             termSideCondition =
                 sideCondition `SideCondition.andCondition` simplifiedCondition'
-        simplifiedTerm <- simplifyConditionalTerm termSideCondition term'
+        simplifiedSideCondition <-
+            simplifySideCondition termSideCondition
+        simplifiedTerm <- simplifyConditionalTerm simplifiedSideCondition term'
         let simplifiedPattern =
                 Conditional.andCondition simplifiedTerm simplifiedCondition
         simplifyCondition sideCondition simplifiedPattern
+
+-- | Simplify each condition in the 'SideCondition' with the assumption that
+-- the other conditions are true.
+simplifySideCondition
+    :: forall variable simplifier
+    .  InternalVariable variable
+    => MonadSimplify simplifier
+    => SideCondition variable
+    -> LogicT simplifier (SideCondition variable)
+simplifySideCondition (toList . from @_ @(MultiAnd _) -> predicates) =
+    State.execStateT (worker predicates) SideCondition.top
+  where
+    worker
+        :: [Predicate variable]
+        -> StateT
+            (SideCondition variable)
+            (LogicT simplifier)
+            ()
+    worker [] = return ()
+    worker (pred' : rest) = do
+        sideCondition <- State.get
+        let otherConds = sideCondition <> from (MultiAnd.make rest)
+        result <-
+            simplifyCondition otherConds (Condition.fromPredicate pred')
+            & lift
+        State.put (SideCondition.andCondition sideCondition result)
+        worker rest
```

Comment:
Let me make sure I understand this correctly: The `SideCondition` comprised of all simplified `Predicate`s is held in the `State`. At each step, we combine that with the `SideCondition` from all (other) un-simplified `Predicate`s, and use the combination to simplify the current `Predicate`. Then we update the stored `SideCondition`. Is that right?

Reply:
That's right.